src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -252,12 +246,23 @@ public final void unregisterSession(final Session session) {
         } catch (Exception ex) {
             throw new NakadiRuntimeException(ex);
         }
-
-        return loadDataAsync(
-                zkSessions,
-                key -> getSubscriptionPath("/sessions/" + key),
-                this::deserializeSession
-        ).values();
+        Map result = new HashMap();
+        for (int i = 0; i < 3; i++) {
```

Comment:
It makes no sense to query same sessions all the time - if they are changed, we have to get the new data from zk (line 243)